pkg/collector/container_hc_collector.go

```diff
@@ -115,11 +110,16 @@ func (c *Collector) updateBPFMetrics() {
 	foundProcess := make(map[uint64]bool)
 	var ct ProcessBPFMetrics
 	for it := c.bpfHCMeter.Table.Iter(); it.Next(); {
+		key := it.Key()
 		data := it.Leaf()
 		err := binary.Read(bytes.NewBuffer(data), utils.DetermineHostByteOrder(), &ct)
 		if err != nil {
 			klog.V(5).Infof("failed to decode received data: %v", err)
-			continue
+			continue // this only happens if there is a problem in the bpf code
+		}
+		err = c.bpfHCMeter.Table.Delete(key) // deleting the element to reset the counter values
```

Comment:
seems this is different to the proposal in original issue? 
so we delete here one by one instead of use the DeleteALL that trigger a bigger loop? 
but will this get performance improvement?

Reply:
Yes because we use only one loop iteration to read and delete the keys.
The other approaches require to iterate the map again to delete the keys.

Reply:
This is the profiling of the fix on the test environment. 

```console
# go tool pprof _output/bin/linux_amd64/kepler  /tmp/k.626
File: kepler
Build ID: f61fa7c866b152f79fd015df77cb00bf2b7c87b0
Type: cpu
Time: Apr 13, 2023 at 10:22am (EDT)
Duration: 60s, Total samples = 15.87s (26.45%)
Entering interactive mode (type "help" for commands, "o" for options)
(pprof) top20 -cum (all)
Active filters:
   focus=(all)
Showing nodes accounting for 5.55s, 34.97% of 15.87s total
Dropped 223 nodes (cum <= 0.08s)
Showing top 20 nodes out of 155
      flat  flat%   sum%        cum   cum%
         0     0%     0%      2.79s 17.58%  github.com/sustainable-computing-io/kepler/pkg/collector.(*Collector).Update
         0     0%     0%      2.79s 17.58%  github.com/sustainable-computing-io/kepler/pkg/manager.(*CollectorManager).Start.func1
         0     0%     0%      2.62s 16.51%  github.com/sustainable-computing-io/kepler/pkg/collector.(*Collector).updateBPFMetrics
         0     0%     0%      2.57s 16.19%  github.com/containerd/cgroups.(*cgroup).Stat.func1
     2.47s 15.56% 15.56%      2.49s 15.69%  runtime.cgocall
     0.13s  0.82% 16.38%      1.94s 12.22%  runtime.mallocgc
     1.62s 10.21% 26.59%      1.74s 10.96%  syscall.Syscall
         0     0% 26.59%      1.70s 10.71%  os.ReadFile
         0     0% 26.59%      1.59s 10.02%  github.com/containerd/cgroups.readUint
         0     0% 26.59%      1.57s  9.89%  os.Open (inline)
         0     0% 26.59%      1.57s  9.89%  os.OpenFile
         0     0% 26.59%      1.57s  9.89%  os.openFileNolog
         0     0% 26.59%      1.45s  9.14%  runtime.mcall
         0     0% 26.59%      1.43s  9.01%  github.com/iovisor/gobpf/bcc.(*TableIterator).Next
     0.03s  0.19% 26.78%      1.40s  8.82%  runtime.schedule
     0.02s  0.13% 26.91%      1.36s  8.57%  runtime.newobject
         0     0% 26.91%      1.33s  8.38%  syscall.Open (inline)
         0     0% 26.91%      1.33s  8.38%  syscall.openat
     1.28s  8.07% 34.97%      1.29s  8.13%  syscall.Syscall6
         0     0% 34.97%      1.23s  7.75%  github.com/iovisor/gobpf/bcc.(*TableIterator).Next.func5
```

Reply:
the batch operation is introduced in [linux kernel 5.6](https://github.com/iovisor/bcc/blob/master/docs/kernel-versions.md). Some OS, e.g. RHEL, have backported though, so kernel version is not a reliable indicator. 

To be compatible, can you try batch delete and fall back to single delete  if not working?

Reply:
@rootfs I added the logic to use the batch key deletion if the kernel is newer or equal 5.6 and the option that we user can enable it if the OS has backported the bpf feature.

Can you review this PR again?